Make inductor config hashing more portable

oulgen opened the pull request:
Summary: masnesral and I noticed that config contains non portable artifacts. Lets fix that.

Test Plan: adhoc testing

Differential Revision: D57748025




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

masnesral commented on torch/_inductor/codecache.py:
```diff
@@ -656,11 +656,11 @@ def __init__(
 
         try:
-            self.inductor_config = config.save_config()
+            self.inductor_config = config.save_config_for_cache()
```
nit: Naming this "save_config_for_hash()" might be slightly more accurate? Just mentioning; I don't have a strong opinion.

masnesral commented on torch/utils/_config_module.py:
```diff
@@ -156,6 +156,19 @@ def save_config(self) -> bytes:
             config.pop(key)
         return pickle.dumps(config, protocol=2)
 
+    def save_config_for_cache(self) -> bytes:
```
It occurs to me that we could stick the whole dict into a field in the FxGraphDetails object rather than pickling it here and storing the serialized bytes in the details. The only advantage to doing that is the debug output we were inspecting today would show each field separately instead of just calling it "bytes"

oulgen committed:
Make inductor config hashing more portable (#127022)

Summary:

masnesral and I noticed that config contains non portable artifacts. Lets fix that.

Test Plan: adhoc testing

Reviewed By: masnesral

Differential Revision: D57748025
